app/views/contexts/summary/question.py

```diff
@@ -95,10 +108,58 @@ def _build_answers(
             ]
 
         summary_answers = []
+
+        if self.dynamic_answer_schemas:
+            for answer_schema in self.dynamic_answer_schemas:
+                for list_item_id in self.list_store.get(
+                    self.dynamic_answer_identifier
+                ).items:
+                    placeholder_renderer = PlaceholderRenderer(
```

Comment:
Could we not resolve the schema upfront then iterate over it, then we can handle dynamic/static with 1 loop?.

Reply:
So this one is refactored now, it is resolved before `for schema in schemas` loop.